mlrose_hiive/fitness/queens.py

```diff
@@ -96,3 +99,11 @@ def get_prob_type(self):
             or 'either'.
         """
         return self.prob_type
+
+    @staticmethod
+    def get_max_size(problem_size):
```

Comment:
I have made the following upper bound:
If length is 1, 0,
if length 2, 1,
for length 3 or higher, it is 3 * (length-2)
This can be proved by restating the fitness function in terms of matching queens to the right of any queen (as matches on the left of one queen is on the right of the matching queen). For each queen you can match on any of the 2 diagonals or row, so at most 3 matches. The reason we have length-2, for the most right queen it can't match, for the second most right queen it can match at most 1 queen, for the 3rd most right queen it can match at most 2 queens. Putting this all together we get 3*(length-3) + 2 + 1 = 3*(length - 2).